SQLAssignment.sql.sql

```diff
@@ -0,0 +1,1043 @@
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3002 | 1900.10 | 1990-10-03 | 2007 | 1004 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3008 | 4723.00 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
++------+---------+------------+------+------+
+  
+17) Write a query that uses the EXISTS operator to extract all salespeople who have customers with a rating of 300.
+
+ SELECT SNUM,SNAME,CITY,COMM 
+ FROM Salespeople s WHERE EXISTS (SELECT SNUM,RATING  
+                                  FROM Customers c WHERE c.SNUM = s.SNUM AND RATING = 300);
++------+--------+-----------+------+
+| SNUM | SNAME  | CITY      | COMM |
++------+--------+-----------+------+
+| 1002 | Serres | SanJose   |   13 |
+| 1007 | Rifkin | Barcelona |   15 |
++------+--------+-----------+------+
+18) Find all customers whose cnum is 1000 above the snum of Serres.
+
+mysql> SELECT Customers.CNUM,Customers.CNAME 
+	   FROM Customers,Salespeople 
+       WHERE Customers.SNUM = Salespeople.SNUM AND Customers.CNUM > Salespeople.SNUM+1000;
++------+----------+
+| CNUM | CNAME    |
++------+----------+
+| 2006 | Clemens  |
+| 2003 | Liu      |
+| 2004 | Grass    |
+| 2007 | Pereira  |
+| 2008 | Cisneros |
++------+----------+
+19) Give the salespeople’s commissions as percentages instead of decimal numbers.
+mysql> SELECT CONCAT (COMM,'%') AS PERCENTAGE FROM Salespeople;
++------------+
+| PERCENTAGE |
++------------+
+| 12%        |
+| 13%        |
+| 10%        |
+| 11%        |
+| 15%        |
+| 25%        |
++------------+
+20) Find the largest order taken by each salesperson on each date, eliminating those Maximum orders, which are less than 3000.
+mysql> SELECT o.SNUM,o.ODATE,SUM(O.AMT) AS Largest_Orders FROM Orders o,Salespeople s WHERE  o.SNUM =s.SNUM AND o.AMT > 3000 GROUP BY o.ODATE;
++------+------------+----------------+
+| SNUM | ODATE      | Largest_Orders |
++------+------------+----------------+
+| 1002 | 1990-10-03 |        5160.45 |
+| 1001 | 1990-10-05 |        4723.00 |
+| 1001 | 1990-10-06 |        9891.88 |
++------+------------+----------------+
+21) List all the largest orders for October 3, for each salesperson.
+mysql> SELECT o.SNUM,o.ODATE,SUM(O.AMT) AS Largest_Orders FROM Orders o,Salespeople s WHERE  o.SNUM =s.SNUM AND o.ODATE = ' 1990-10-03 ' GROUP BY o.SNUM;
++------+------------+----------------+
+| SNUM | ODATE      | Largest_Orders |
++------+------------+----------------+
+| 1007 | 1990-10-03 |        1116.85 |
+| 1001 | 1990-10-03 |         767.19 |
+| 1004 | 1990-10-03 |        1900.10 |
+| 1002 | 1990-10-03 |        5160.45 |
++------+------------+----------------+
+
+22) Find all customers located in cities where Serres has customers.
+mysql> SELECT CNUM,CNAME,CITY FROM Customers WHERE SNUM IN (SELECT SNUM FROM Salespeople WHERE SNUM = 1002);
```

Comment:
You should put conditional check for salesperson name as "Serres" instead of using SNUM of that salesperson. 
>> SELECT SNUM FROM Salespeople WHERE SNUM = 1002